onnxruntime/core/platform/windows/env.cc

```diff
@@ -137,12 +137,21 @@ class WindowsThread : public EnvThread {
   static unsigned __stdcall ThreadMain(void* param) {
     std::unique_ptr<Param> p(static_cast<Param*>(param));
 
-    const ORTCHAR_T* name_prefix =
-        (p->name_prefix == nullptr || wcslen(p->name_prefix) == 0) ? L"onnxruntime" : p->name_prefix;
-    std::wostringstream oss;
-    oss << name_prefix << "-" << p->index;
-    // Ignore the error
-    (void)SetThreadDescription(GetCurrentThread(), oss.str().c_str());
+    // Not all machines have SetThreadDescription (e.g. Azure App Service sandbox) so we need to ensure
+    // it's available before calling.
+    HMODULE kernelModule = GetModuleHandle(TEXT("kernel32.dll"));
```

Comment:
kernel32.dll does not always exist. Our nuget packages are built for OneCore. Some OneCore editions do not have kernel32.dll. And, you may fetch a prebuilt onnxruntime.dll from the packages, you will find it does not depend on kernel32.dll.

Reply:
So, before PR #15219, `WINVER >= _WIN32_WINNT_WIN10` is false.  `WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)` is also false. 
It went directly to `constexpr SetThreadDescriptionFunc pSetThrDesc = nullptr;`

Reply:
We can skip it altogether if this becomes difficult.

Reply:
Agree. It doesn't bring much value.

Reply:
I can see it would be helpful when investigating issues as it you can see which threads are intra-op vs inter-op etc. when this works. I'll change it to check if GetModuleHandle returns a value.

Validated this change works with a test Azure App Services test app.

Reply:
Then, I suggest adding the guard `WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)` back since we know for sure that the function is only in desktop apps and UWP apps.